packages/contracts/contracts/voting/token/TokenVoting.sol

```diff
@@ -64,13 +69,14 @@ contract TokenVoting is MajorityVotingBase {
         VoteOption _voteOption,
         bool _tryEarlyExecution
     ) external override returns (uint256 proposalId) {
-        uint64 snapshotBlock;
+        uint256 snapshotBlock;
         unchecked {
-            snapshotBlock = block.number.toUint64() - 1;
+            snapshotBlock = block.number - 1;
         }
 
-        uint256 totalVotingPower = votingToken.getPastTotalSupply(snapshotBlock);
-        if (totalVotingPower == 0) revert NoVotingPower();
+        if (totalVotingPower(uint256(snapshotBlock)) == 0) {
```

Comment:
no need to convert it again.

Reply:
Fixed in bab7649c868ddc5de9cde6fd07ee7839fac9853a. It happened because of all the back and forth we are doing